addon/services/sticky-manager.js

```diff
@@ -0,0 +1,245 @@
+       * a difference between how the throttle implementations work
+       * between the React and Ember worlds, which meant that sticky
+       * items in Ember could end up in the wrong position after a
+       * scroll/resize event, until the next such event. This is a
+       * workaround that ensures that no sticky items will be left
+       * in the wrong positions more than 50ms after these events.
+       */
+      this.debounceTask('manageStickyItems', 50);
+    },
+
+    evaluateStickyItem(stickyItem, scrollTop, containerTop) {
+      let {
+        stickyNode,
+        placeHolderNode,
+        boundingElement,
+        offset,
+        disableWhenStacked,
+      } = stickyItem;
+
+      if (disableWhenStacked && stackedContent().matches) {
+        return {
+          sticky: false,
+          top: 0,
+          left: 0,
+          width: 'auto',
+        };
+      }
+
+      let stickyOffset = offset
+        ? this.getOffset(stickyNode) + parseInt(tokens.spacingLoose, 10)
+        : this.getOffset(stickyNode);
+
+      let scrollPosition = scrollTop + stickyOffset;
+      let placeHolderNodeCurrentTop =
+        getRectForNode(placeHolderNode).top - containerTop + scrollTop;
+      let top = containerTop + stickyOffset;
+      let width = getRectForNode(placeHolderNode).width;
+      let left = getRectForNode(placeHolderNode).left;
+
+      let sticky;
+
+      if (boundingElement == null) {
+        sticky = scrollPosition >= placeHolderNodeCurrentTop;
+      } else {
+        let stickyItemHeight = getRectForNode(stickyNode).height;
+        let stickyItemBottomPosition =
+          getRectForNode(boundingElement).bottom -
+          stickyItemHeight +
+          scrollTop -
+          containerTop;
+
+        sticky =
+          scrollPosition >= placeHolderNodeCurrentTop &&
+          scrollPosition < stickyItemBottomPosition;
+      }
+
+      return {
+        sticky,
+        top,
+        left,
+        width,
+      };
+    },
+
+    updateStuckItems(item, sticky) {
+      let { stickyNode } = item;
+      if (sticky && !this.isNodeStuck(stickyNode)) {
+        this.addStuckItem(item);
+      } else if (!sticky && this.isNodeStuck(stickyNode)) {
+        this.removeStuckItem(item);
+      }
+    },
+
+    addStuckItem(stickyItem) {
+      this.get('stuckItems').push(stickyItem);
+    },
+
+    removeStuckItem(stickyItem) {
+      let stuckItems = this.get('stuckItems');
+      let { stickyNode: nodeToRemove } = stickyItem;
+      let nodeIndex = stuckItems.findIndex(
+        ({ stickyNode }) => nodeToRemove === stickyNode
+      );
+      stuckItems.splice(nodeIndex, 1);
+    },
+
+    getOffset(node) {
+      let stuckItems = this.get('stuckItems');
+      let stuckItemsLength = stuckItems.get('length');
+      if (stuckItemsLength === 0) {
+        return 0;
+      }
+
+      let offset = 0;
+      let count = 0;
+      let stuckNodesLength = stuckItemsLength;
```

Comment:
any reason for not just declaring `stuckItemsLength` as `stuckNodesLength` and drop this?

Reply:
Ah this is just my bad 😛